src/test/java/com/commercetools/sync/commons/CleanupTest.java

```diff
@@ -0,0 +1,105 @@
+package com.commercetools.sync.commons;
+
+import com.commercetools.sync.commons.models.FetchCustomObjectsGraphQlRequest;
+import com.commercetools.sync.commons.models.ResourceKeyId;
+import com.commercetools.sync.commons.models.ResourceKeyIdGraphQlResult;
+import io.sphere.sdk.client.BadRequestException;
+import io.sphere.sdk.client.NotFoundException;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.commands.CustomObjectDeleteCommand;
+import io.sphere.sdk.utils.CompletableFutureUtils;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class CleanupTest {
+    private final SphereClient mockClient = mock(SphereClient.class);
+    private static final int deleteDaysAfterLastModification = 30;
+
+    @Test
+    void deleteUnresolvedReferences_withDeleteDaysAfterLastModification_ShouldDeleteAndReturnCleanupStatistics() {
+        final ResourceKeyIdGraphQlResult resourceKeyIdGraphQlResult = mock(ResourceKeyIdGraphQlResult.class);
+        when(resourceKeyIdGraphQlResult.getResults()).thenReturn(new HashSet<>(Arrays.asList(
+            new ResourceKeyId("coKey1", "coId1"),
+            new ResourceKeyId("coKey2", "coId2"))));
+
+        when(mockClient.execute(any(FetchCustomObjectsGraphQlRequest.class)))
+            .thenReturn(CompletableFuture.completedFuture(resourceKeyIdGraphQlResult));
+
+        when(mockClient.execute(any(CustomObjectDeleteCommand.class)))
+            .thenReturn(CompletableFuture.completedFuture(mock(CustomObject.class)));
+
+        final Cleanup.Statistics statistics =
+            Cleanup.of(mockClient).deleteUnresolvedReferenceCustomObjects(deleteDaysAfterLastModification)
+                   .join();
+
+        assertThat(statistics.getTotalDeleted()).isEqualTo(4);
+        assertThat(statistics.getTotalFailed()).isEqualTo(0);
+        assertThat(statistics.getReportMessage())
+            .isEqualTo("Summary: 4 custom objects were deleted in total (0 failed to delete).");
+        assertThat(statistics.getTimeElapsedInMilliseconds()).isLessThan(1000);
```

Comment:
```suggestion
        assertThat(statistics.getTimeElapsedInMilliseconds()).isLessThan(1000).isGreaterThan(0);
```

Reply:
I remove this from statistics which could be done easily by the library user if it's needed.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/651/commits/d76fec40ae38d5d9656c272d3706249cc5c8bdb0